include/zephyr/posix/posix_types.h

```diff
@@ -47,7 +47,7 @@ struct pthread_attr {
 	int32_t detachstate;
 	uint32_t initialized;
 };
-#if defined(CONFIG_MINIMAL_LIBC) || defined(CONFIG_PICOLIBC)
+#if defined(CONFIG_MINIMAL_LIBC) || defined(CONFIG_PICOLIBC) || defined(CONFIG_ARMCLANG_STD_LIBC)
```

Comment:
Should we flip this logic so that it excludes newlib rather than including every other C library? Can @cfriedt explain why the structs are always defined but the typedef is only included for known C libraries? It seems like the struct shouldn't be used in the newlib case either...

Reply:
Newlib already includes a typedef for this, and the type that newlib defines is larger than the structure that zephyr defines. Given that the plan for POSIX is to blow some of this away and replace with more mature POSIX headers, I'm fine with this change for now.

Reply:
Yeah, I was just asking if this should be `#ifndef CONFIG_NEWLIB_LIBC` instead of listing all other C libraries. As you say, it's transient, so it probably doesn't matter.

Reply:
@galak - Keith has a good point there. Could you add another commit that modifies `tests/posix/common/testcase.yaml`?